File: GFG/Maximize the sum of selected numbers from a sorted array to make it empty/Maximize the sum of selected numbers from a sorted array to make it empty.py
class Solution:
    def maximizeSum (self,arr, n) :
        d = dict()
        for i in arr:
            d[i] = d.get(i, 0) + 1
        ans = 0
        for i in range(n - 1, -1, -1):
            if d[arr[i]] <= 0: continue
                
            if arr[i] - 1 in d and d[arr[i] - 1] > 0:
                ans += arr[i]
                d[arr[i] - 1] -= 1
                d[arr[i]] -= 1
                
        for i in d:
            ans += d[i] * i
        return ans

File: GFG/Maximize the sum of selected numbers from an array to make it empty/Maximize the sum of selected numbers from a sorted array to make it empty.cpp
class Solution{
    public:
    int maximizeSum(int a[], int n) 
    {
        // Complete the function
        
          unordered_map<int, int> ans;
        int sum=0;
        bool arr[n];
        for(int i=0;i<n;i++){
            arr[i]=false;
        }
        for (int i = 0; i < n; i++)
        	ans[a[i]]++;
        
        for(int i=n-1;i>=0;i--)
        {
            if(arr[i] == false)
            {
                arr[i]=true;
                sum+= a[i];
                if(ans[a[i]-1] > 0)
                {
                    arr[i-ans[a[i]]]=true;
                    ans[a[i]-1]--;
                }
            }
        }
        return sum;
    }

};
